staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -84,27 +62,21 @@ func WithPriorityAndFairness(
 			return
 		}
 
-		var classification *PriorityAndFairnessClassification
-		note := func(fs *fcv1a1.FlowSchema, pl *fcv1a1.PriorityLevelConfiguration) {
-			classification = &PriorityAndFairnessClassification{
-				FlowSchemaName:    fs.Name,
-				FlowSchemaUID:     fs.UID,
-				PriorityLevelName: pl.Name,
-				PriorityLevelUID:  pl.UID}
-		}
+		classification := &utilflowcontrol.PriorityAndFairnessClassification{}
+		ctx = utilflowcontrol.WithClassification(ctx, classification)
+		defer utilflowcontrol.WriteClassificationToResponseTrailerHeaders(ctx, w)()
+
 		var served bool
 		execute := func() {
 			served = true
-			innerCtx := context.WithValue(ctx, priorityAndFairnessKey, classification)
-			innerReq := r.Clone(innerCtx)
+			innerReq := r.Clone(ctx)
```

Comment:
Why not add the headers to `w` here, instead of making them trailers later?

Reply:
reverted the change